prettypandas/styler.py

```diff
@@ -150,8 +160,21 @@ def multi_summary(self, funcs, titles, axis=0, **kwargs):
             return self.multi_summary(funcs, titles, axis=0, **kwargs)\
                        .multi_summary(funcs, titles, axis=1, **kwargs)
 
-        output = [self.data.apply(f, axis=axis, **kwargs).to_frame(t)
-                  for f, t in zip(funcs, titles)]
+        output = []
+        if axis == 0:
+            df = self.data.transpose() #use df to iterate over the rows of the transpose of self.data (i.e. cols of self.data)
+        else:
+            df = self.data
+            
+        if exclude is not None and subset is None:
+            subset = [n for n in list(df.index) if n not in exclude]
+            
+        for f, t in zip(funcs, titles):
+            if subset is None:
+                output.append(self.data.apply(f, axis=axis, **kwargs).to_frame(t)) #apply returns Series, to_frame converts to DataFrame
+            elif subset is not None:
+                summary_vals = [f(vals, **kwargs) if item_name in subset else None for item_name, vals in df.iterrows()]
+                output.append(pd.DataFrame(data = {t: summary_vals}, index = df.index))  #dataframe with column name t and values summary_vals
```

Comment:
I'd prefer this project use pep8 with 79 char column width. I'd prefer this formatted like this:

```python
        output = []
        if axis == 0:
            df = self.data.transpose()  # iterate over cols
        else:
            df = self.data

        if exclude is not None and subset is None:
            subset = [n for n in list(df.index) if n not in exclude]

        for f, t in zip(funcs, titles):
            if subset is None:
                output.append(self.data.apply(f, axis=axis, **kwargs)
                                       .to_frame(t))
            elif subset is not None:
                summary_vals = [f(vals, **kwargs) if item_name in subset
                                else None
                                for item_name, vals in df.iterrows()]

                output.append(pd.DataFrame(data={t: summary_vals},
                                           index=df.index))

```